### PR TITLE
tests: Make e2e tests call the backend

### LINKED ISSUE #3575: tests: Make e2e tests setup their own tests databases
### Description and goal

`e2e` tests are currently intercepting and mocking calls to the backend. Cleaning of integration tests has shown that e2e was relying on leftovers of the integration tests to work: (see this required fix: https://github.com/DGEXSolutions/osrd/pull/3548/commits/50b07f94e19ab68aa058d3ce93e5a3f37a6fae9b).

We should find a way to use pytest fixtures to setup a clean test environment for e2e tests.

### Acceptance criteria

e2e tests should work without `intercept` at all

### PR DESCRIPTION
**Pull request by bloussou**
:warning: As this PR is running e2e tests as "real e2e tests" it needs to setup and cleanup data (so at least some python/backend  notion),

This PR is :
- running integration tests using pytest
- removing intercepts
- changing rolling stocks `data-testid` from `id` to rolling stock names (it is the key in the database so it's ok)
- adding a fixture to create multiple rolling stocks
- making sure that the database is empty after tests are run

:next_track_button: A follow up of this pr is to make sure that tests can be run on a non empty database (explicit opening of projects and selection of rolling stocks.)

close #3575

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -82,23 +82,35 @@ def small_scenario(small_infra: Infra, foo_project_id: int, foo_study_id: int) -
...
-    response = requests.post(f"{API_URL}/rolling_stock/", json=json.loads(FAST_ROLLING_STOCK_JSON_PATH.read_text()))
-    if response.status_code == 200:
-        return response.json()["id"]
-    # Rolling stock already exists
-    response = requests.get(f"{API_URL}rolling_stock/?page_size=1000")
-    rolling_stocks = response.json()["results"]
-    return next(
-        iter([rolling_stock["id"] for rolling_stock in rolling_stocks if rolling_stock["name"] == "fast_rolling_stock"])
+def _create_fast_rolling_stocks(
+    names: Iterable[Optional[str]] = [None], metadata: Iterable[Optional[Mapping[str, str]]] = [None]
```
Comment (eckter) [inline tests/conftest.py]: List literals as default arguments is risky, and pycharm complains about it. Is there a reason not to use `None` directly?

`Iterable` don't support the `[]` operator, the type should probably be `List` if it's what's expected

**Review by eckter — COMMENTED** [review]
There's one nitpicky comment, but other than that, LGTM

(I only reviewed the python part)

File: tests/conftest.py (hunk 1 above)
Comment (bloussou) [reply tests/conftest.py]: Indeed it was not clean, I have changed it for `names_and_metadata: Optional[Mapping[str, Mapping[str, str]]] = None` :smile:

**Review by eckter — DISMISSED** [review]

**Review by flomonster — DISMISSED** [review]
LGTM

File: front/src/common/RollingStockSelector/RollingStockCard.jsx [hunk 2]
```diff
@@ -55,7 +55,7 @@ function RollingStockCard(props) {
       onClick={displayCardDetail}
       tabIndex={0}
       ref={ref2scroll}
-      data-testid={`rollingstock${data.id}`}
```
Comment (Uriel-Sautron) [inline front/src/common/RollingStockSelector/RollingStockCard.jsx:58]: Why not keep a more meaningful data-testid with "rollingstock" before the data.name.
`data-testid={rollingstock-${data.name}}`

Comment (bloussou) [reply front/src/common/RollingStockSelector/RollingStockCard.jsx:58]: Indeed :)

**Review by eckter — APPROVED** [review]

**Review by clarani — APPROVED** [review]
It would be nice to add a command in the front to launch the new e2e tests :smile: 

Thanks for this PR !

**Comment by bloussou** [comment]
Thanks @clarani :) I will do this after it is merged I have opened a ticket https://github.com/DGEXSolutions/osrd/issues/3760

**Review by Math-R — APPROVED** [review]

**Review by flomonster — APPROVED** [review]

**Pull request merged**